server/server.go

```diff
@@ -103,8 +103,9 @@ type Server struct {
 type pullArguments struct {
 	image         string
 	sandboxCgroup string
-	credentials   imageTypes.DockerAuthConfig
+	credentials   imageTypes.DockerAuthConfig // parsed auth configuration
 	namespace     string
+	auth          *types.AuthConfig // original auth configuration
```

Comment:
See how `s.pullOperationsInProgress` is indexed, all of `pullArguments` is compared _by value_.

Including a pointer into the value, one that is quite likely to be different on each operation, AFAICS completely breaks this, because no new pull request is likely to match any existing `pullOperationsInProgress` member.

Reply:
Hm, yeah I removed support for auth for now, I think we need a different way to make it work.